// transcend-react/lib/components/ApplicationSideBar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ReactCSSTransitionGroup = require('react/lib/ReactCSSTransitionGroup');

var _ReactCSSTransitionGroup2 = _interopRequireDefault(_ReactCSSTransitionGroup);

var _CompanySelectorPanel = require('./CompanySelectorPanel');

var _CompanySelectorPanel2 = _interopRequireDefault(_CompanySelectorPanel);

var _UserMenu = require('./UserMenu');

var _UserMenu2 = _interopRequireDefault(_UserMenu);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ApplicationSideBar = function (_Component) {
  (0, _inherits3.default)(ApplicationSideBar, _Component);

  function ApplicationSideBar() {
    var _ref;

    (0, _classCallCheck3.default)(this, ApplicationSideBar);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = ApplicationSideBar.__proto__ || (0, _getPrototypeOf2.default)(ApplicationSideBar)).call.apply(_ref, [this].concat(args)));

    _this.state = {};
    _this.onCollapserClick = _this.onCollapserClick.bind(_this);
    _this.onCompanyClick = _this.onCompanyClick.bind(_this);
    _this.onCompanySelectorPanelHide = _this.onCompanySelectorPanelHide.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(ApplicationSideBar, [{
    key: 'onCollapserClick',
    value: function onCollapserClick(evt) {
      evt.preventDefault();
      this.props.onResize(!this.props.collapsed);
    }
  }, {
    key: 'onCompanyClick',
    value: function onCompanyClick(evt) {
      evt.preventDefault();
      this.setState({
        showCompany: !this.state.showCompany
      });
    }
  }, {
    key: 'onCompanySelectorPanelHide',
    value: function onCompanySelectorPanelHide() {
      this.setState({
        showCompany: false
      });
    }
  }, {
    key: 'getClassName',
    value: function getClassName() {
      var base = 'application-side-bar';
      var className = [base];
      if (this.props.className) {
        className.push(this.props.className);
      }
      if (this.props.collapsed) {
        className.push(base + '--collapsed');
      }
      if (this.props.user) {
        className.push(base + '--has-user');
      }
      if (this.state.showCompany) {
        className.push(base + '--showing-company-selector');
      }
      return className.join(' ');
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { className: this.getClassName() },
        _react2.default.createElement('div', { className: 'application-side-bar__right-border' }),
        _react2.default.createElement(
          'div',
          { className: 'application-side-bar__flex' },
          this.renderCurrentCompany(),
          _react2.default.createElement(
            'nav',
            { className: 'application-side-bar__nav' },
            _react2.default.createElement(
              'ul',
              { className: 'application-side-bar__nav-list' },
              this.renderChildren()
            )
          ),
          _react2.default.createElement(
            'div',
            { className: 'application-side-bar__footer' },
            this.renderUser(),
            this.renderFooterSplit()
          )
        )
      );
    }
  }, {
    key: 'renderChildren',
    value: function renderChildren() {
      return _react2.default.Children.map(this.props.children, function (n) {
        return _react2.default.createElement(
          'li',
          { className: 'application-side-bar__nav-item' },
          n
        );
      });
    }
  }, {
    key: 'renderFooterSplit',
    value: function renderFooterSplit() {
      return _react2.default.createElement(
        'div',
        { className: 'application-side-bar__footer__flex' },
        _react2.default.createElement(
          'a',
          {
            tabIndex: '0',
            className: 'application-side-bar__collapser',
            onClick: this.onCollapserClick
          },
          _react2.default.createElement('i', null)
        ),
        _react2.default.createElement(
          'div',
          { className: 'application-side-bar__logo' },
          _react2.default.createElement('i', { style: this.logoStyles() })
        )
      );
    }
  }, {
    key: 'renderUser',
    value: function renderUser() {
      if (this.props.user) {
        return _react2.default.createElement(
          'div',
          {
            className: 'application-side-bar__user'
          },
          _react2.default.createElement(_UserMenu2.default, { user: this.props.user, showSupport: !this.isWhiteLabel(), csrfToken: this.props.csrfToken })
        );
      }
      return undefined;
    }
  }, {
    key: 'renderCurrentCompany',
    value: function renderCurrentCompany() {
      if (this.props.currentCompany) {
        if (this.shouldRenderSelectorPanel()) {
          return this.renderCurrentCompanySelectorPanel();
        }
        return _react2.default.createElement(
          'div',
          { className: 'application-side-bar__company' },
          _react2.default.createElement(
            'div',
            { className: 'application-side-bar__company__container' },
            _react2.default.createElement(
              'div',
              { key: 'logo', className: 'application-side-bar__company__logo' },
              _react2.default.createElement('img', { src: this.props.currentCompany.fav_icon_url, role: 'presentation', alt: this.props.currentCompany.name })
            ),
            _react2.default.createElement(
              'div',
              { key: 'name', className: 'application-side-bar__company__name' },
              this.props.currentCompany.name
            )
          )
        );
      }
      return undefined;
    }
  }, {
    key: 'renderCurrentCompanySelectorPanel',
    value: function renderCurrentCompanySelectorPanel() {
      return _react2.default.createElement(
        'div',
        { className: 'application-side-bar__company' },
        _react2.default.createElement(
          'a',
          {
            tabIndex: '0',
            className: 'application-side-bar__company__container application-side-bar__company__container--clickable',
            onClick: this.onCompanyClick
          },
          _react2.default.createElement(
            'div',
            { className: 'application-side-bar__company__logo' },
            this.renderLogo()
          ),
          _react2.default.createElement(
            'div',
            { className: 'application-side-bar__company__name' },
            this.props.currentCompany.name
          ),
          _react2.default.createElement(
            'div',
            { className: 'application-side-bar__company__arrow' },
            _react2.default.createElement('span', { className: 'icon--chevron-down text--xxsmall bold' })
          )
        ),
        _react2.default.createElement(
          _ReactCSSTransitionGroup2.default,
          {
            component: 'div',
            transitionName: 'fadeIn',
            transitionEnter: true,
            transitionEnterTimeout: 200,
            transitionLeaveTimeout: 200
          },
          this.renderCompanySelectorPanel()
        )
      );
    }
  }, {
    key: 'renderCompanySelectorPanel',
    value: function renderCompanySelectorPanel() {
      if (this.state.showCompany && this.props.currentCompany) {
        return _react2.default.createElement(_CompanySelectorPanel2.default, {
          currentCompany: this.props.currentCompany,
          companies: this.props.companies,
          user: this.props.user,
          location: this.props.location,
          params: this.props.params,
          onHide: this.onCompanySelectorPanelHide,
          showCreateCompany: this.canCreateCompanies()
        });
      }
      return undefined;
    }
  }, {
    key: 'renderLogo',
    value: function renderLogo() {
      var firstInitial = this.props.currentCompany.name.slice(0, 1);
      if (this.props.currentCompany.navbar_logo_url) {
        return _react2.default.createElement('img', { src: this.props.currentCompany.navbar_logo_url, role: 'presentation', alt: this.props.currentCompany.name });
      }
      return _react2.default.createElement(
        'div',
        { className: 'selector-panel__item__initial' },
        firstInitial
      );
    }
  }, {
    key: 'logoStyles',
    value: function logoStyles() {
      if (this.props.currentCompany && this.props.currentCompany.navbar_logo_url) {
        return {
          backgroundImage: this.props.currentCompany.navbar_logo_url
        };
      }
      return undefined;
    }
  }, {
    key: 'isWhiteLabel',
    value: function isWhiteLabel() {
      return this.props.currentCompany && this.props.currentCompany.is_white_label;
    }
  }, {
    key: 'shouldRenderSelectorPanel',
    value: function shouldRenderSelectorPanel() {
      return this.props.companies && this.props.companies.length || this.canCreateCompanies();
    }
  }, {
    key: 'canCreateCompanies',
    value: function canCreateCompanies() {
      return this.props.currentPosition && this.props.currentPosition.can_manage_agency;
    }
  }]);
  return ApplicationSideBar;
}(_react.Component);

ApplicationSideBar.propTypes = {
  user: _react.PropTypes.object,
  companies: _react.PropTypes.array,
  csrfToken: _react.PropTypes.string,
  currentCompany: _react.PropTypes.object,
  currentPosition: _react.PropTypes.object,
  children: _react.PropTypes.node,
  location: _react.PropTypes.object.isRequired,
  params: _react.PropTypes.object.isRequired,
  onResize: _react.PropTypes.func.isRequired,
  collapsed: _react.PropTypes.bool,
  className: _react.PropTypes.string
};
exports.default = ApplicationSideBar;
