// transcend-react/lib/components/Checkbox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function (_Component) {
  (0, _inherits3.default)(Checkbox, _Component);

  function Checkbox(props) {
    var _ref;

    (0, _classCallCheck3.default)(this, Checkbox);

    for (var _len = arguments.length, args = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = Checkbox.__proto__ || (0, _getPrototypeOf2.default)(Checkbox)).call.apply(_ref, [this, props].concat(args)));

    _this.uniqueId = props.id || _lodash2.default.uniqueId('checkbox');
    return _this;
  }

  (0, _createClass3.default)(Checkbox, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          value = _props.value,
          className = _props.className,
          onClick = _props.onClick,
          onChange = _props.onChange,
          checked = _props.checked,
          defaultChecked = _props.defaultChecked,
          disabled = _props.disabled,
          name = _props.name,
          label = _props.label;

      return _react2.default.createElement(
        'div',
        { className: 'checkbox ' + (className || '') },
        _react2.default.createElement('input', {
          type: 'checkbox',
          id: this.uniqueId,
          name: name,
          value: value,
          onClick: onClick,
          onChange: onChange,
          ref: function ref(n) {
            return _this2.input = n;
          },
          defaultChecked: defaultChecked,
          checked: checked,
          disabled: disabled
        }),
        _react2.default.createElement(
          'label',
          { htmlFor: this.uniqueId },
          label
        )
      );
    }
  }]);
  return Checkbox;
}(_react.Component);

Checkbox.propTypes = {
  checked: _react.PropTypes.bool,
  className: _react.PropTypes.string,
  defaultChecked: _react.PropTypes.bool,
  disabled: _react.PropTypes.bool,
  id: _react.PropTypes.string,
  name: _react.PropTypes.string,
  onClick: _react.PropTypes.func,
  onChange: _react.PropTypes.func,
  label: _react.PropTypes.any,
  value: _react.PropTypes.string
};
exports.default = Checkbox;
