// transcend-react/lib/components/CheckboxGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _typeof2 = require('babel-runtime/helpers/typeof');

var _typeof3 = _interopRequireDefault(_typeof2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CheckboxGroup = function (_React$Component) {
  (0, _inherits3.default)(CheckboxGroup, _React$Component);

  function CheckboxGroup(props, context) {
    (0, _classCallCheck3.default)(this, CheckboxGroup);

    var _this = (0, _possibleConstructorReturn3.default)(this, (CheckboxGroup.__proto__ || (0, _getPrototypeOf2.default)(CheckboxGroup)).call(this, props, context));

    _this.state = {
      items: props.items
    };
    _this.onCheckboxChange = _this.onCheckboxChange.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(CheckboxGroup, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      this.setState({
        items: nextProps.items
      });
    }
  }, {
    key: 'onCheckboxChange',
    value: function onCheckboxChange(event) {
      var index = _lodash2.default.findIndex(this.state.items, function (option) {
        return option.value === event.target.value;
      });

      if (index === -1) {
        return;
      }

      var option = this.state.items[index];
      option.checked = event.target.checked;

      var items = this.state.items;
      items[index] = option;

      this.setState({
        items: items
      });

      this.props.onChange(items);
    }
  }, {
    key: 'render',
    value: function render() {
      var extraClassName = this.props.layoutType === 'inline' ? 'checkbox-group--inline' : '';
      return _react2.default.createElement(
        'div',
        { className: 'checkbox-group ' + extraClassName },
        this.renderCheckboxes()
      );
    }
  }, {
    key: 'renderCheckboxes',
    value: function renderCheckboxes() {
      var _this2 = this;

      return this.props.items.map(function (item, i) {
        var id = 'chck_' + _this2.props.name + '_' + i;
        var text = item;
        var value = item;
        var checked = true;

        if ((typeof item === 'undefined' ? 'undefined' : (0, _typeof3.default)(item)) === 'object') {
          if (item.text) {
            text = item.text;
            if (item.value) {
              value = item.value;
            } else {
              value = text;
            }
          }
          if (item.checked !== true) {
            checked = false;
          }
        }

        return _react2.default.createElement(_Checkbox2.default, {
          id: id,
          key: id,
          value: value,
          checked: checked,
          onChange: _this2.onCheckboxChange,
          label: text
        });
      });
    }
  }]);
  return CheckboxGroup;
}(_react2.default.Component);

CheckboxGroup.propTypes = {
  name: _react2.default.PropTypes.string.isRequired,
  items: _react2.default.PropTypes.array.isRequired,
  onChange: _react2.default.PropTypes.func,
  layoutType: _react2.default.PropTypes.string
};
exports.default = CheckboxGroup;
