// transcend-react/lib/components/CompanySelectorPanelItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _settings = require('../images/settings.svg');

var _settings2 = _interopRequireDefault(_settings);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CompanySelectorPanelItem = function (_React$Component) {
  (0, _inherits3.default)(CompanySelectorPanelItem, _React$Component);

  function CompanySelectorPanelItem() {
    var _ref;

    (0, _classCallCheck3.default)(this, CompanySelectorPanelItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = CompanySelectorPanelItem.__proto__ || (0, _getPrototypeOf2.default)(CompanySelectorPanelItem)).call.apply(_ref, [this].concat(args)));

    _this.onClick = _this.onClick.bind(_this);
    _this.onSettingsClick = _this.onSettingsClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(CompanySelectorPanelItem, [{
    key: 'onClick',
    value: function onClick(evt) {
      evt.preventDefault();
      evt.stopPropagation();
      this.props.onSelect(this.props.item);
    }
  }, {
    key: 'onSettingsClick',
    value: function onSettingsClick(evt) {
      evt.preventDefault();
      evt.stopPropagation();
      this.props.onOptions(this.props.item);
    }
  }, {
    key: 'render',
    value: function render() {
      /* eslint-disable  jsx-a11y/no-static-element-interactions */
      return _react2.default.createElement(
        'li',
        { onClick: this.onClick, className: this.props.className || '' },
        _react2.default.createElement(
          'div',
          { className: 'selector-panel__item__container' },
          this.renderLogo(),
          _react2.default.createElement(
            'div',
            { className: 'selector-panel__item__name company-selector-panel__item__name' },
            this.props.item[this.props.nameAttribute]
          ),
          _react2.default.createElement(
            'a',
            { className: 'selector-panel__item__options company-selector-panel__item__options', onClick: this.onSettingsClick, tabIndex: '0' },
            _react2.default.createElement(_settings2.default, null),
            ' Settings'
          )
        )
      );
    }
  }, {
    key: 'renderLogo',
    value: function renderLogo() {
      var firstInitial = this.props.item[this.props.nameAttribute].slice(0, 1);
      if (this.props.item.navbar_logo_url) {
        return _react2.default.createElement(
          'div',
          { className: 'img-wrap' },
          _react2.default.createElement('img', { src: this.props.item.navbar_logo_url, role: 'presentation' })
        );
      }
      return _react2.default.createElement(
        'div',
        { className: 'selector-panel__item__initial' },
        firstInitial
      );
    }
  }]);
  return CompanySelectorPanelItem;
}(_react2.default.Component);

CompanySelectorPanelItem.propTypes = {
  item: _react2.default.PropTypes.object.isRequired,
  onSelect: _react2.default.PropTypes.func.isRequired,
  onOptions: _react2.default.PropTypes.func.isRequired,
  className: _react2.default.PropTypes.string,
  nameAttribute: _react2.default.PropTypes.string
};
exports.default = CompanySelectorPanelItem;
