// transcend-react/lib/components/ConfirmDialog.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _WaitIndicator = require('./WaitIndicator');

var _WaitIndicator2 = _interopRequireDefault(_WaitIndicator);

var _Modal = require('./Modal');

var _Modal2 = _interopRequireDefault(_Modal);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ConfirmDialog = function (_React$Component) {
  (0, _inherits3.default)(ConfirmDialog, _React$Component);

  function ConfirmDialog() {
    var _ref;

    (0, _classCallCheck3.default)(this, ConfirmDialog);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    /* eslint-disable no-console */
    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = ConfirmDialog.__proto__ || (0, _getPrototypeOf2.default)(ConfirmDialog)).call.apply(_ref, [this].concat(args)));

    console.warn('ConfirmDialog.js is depreciated, use Dialog.js');
    _this.onConfirm = _this.onConfirm.bind(_this);
    _this.onReject = _this.onReject.bind(_this);

    _this.state = {
      displayWaitIndicator: false,
      isDisplayed: false
    };
    return _this;
  }

  (0, _createClass3.default)(ConfirmDialog, [{
    key: 'onConfirm',
    value: function onConfirm() {
      var _this2 = this;

      this.setState({
        displayWaitIndicator: true
      }, function () {
        _this2.props.confirmFunction();
      });
    }
  }, {
    key: 'onReject',
    value: function onReject() {
      var _this3 = this;

      this.setState({
        displayWaitIndicator: true
      }, function () {
        _this3.props.rejectFunction();
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Modal2.default,
        { onShadowClick: this.props.onShadowClick, onCloseClick: this.props.onCloseClick },
        _react2.default.createElement(
          'div',
          { className: 'confirm-dialog dialog' },
          _react2.default.createElement(
            'h1',
            { className: 'confirm-dialog__title' },
            this.props.title
          ),
          _react2.default.createElement(
            'p',
            { className: 'confirm-dialog__text' },
            this.props.text
          ),
          this.props.extraComponent,
          _react2.default.createElement(
            'div',
            { className: 'breathe' },
            this.state.displayWaitIndicator ? _react2.default.createElement(_WaitIndicator2.default, { centered: false }) : this.renderButtons()
          )
        )
      );
    }
  }, {
    key: 'renderButtons',
    value: function renderButtons() {
      return _react2.default.createElement(
        'div',
        { className: 'confirm-dialog__buttons' },
        this.renderSimpleConfirmButton(),
        this.renderRejectButton()
      );
    }
  }, {
    key: 'renderSimpleConfirmButton',
    value: function renderSimpleConfirmButton() {
      return _react2.default.createElement(
        'button',
        { className: 'button ' + (this.props.confirmClassName || ''), onClick: this.onConfirm },
        this.props.confirmText
      );
    }
  }, {
    key: 'renderRejectButton',
    value: function renderRejectButton() {
      if (this.props.rejectText) {
        return _react2.default.createElement(
          'button',
          { className: 'button ' + (this.props.rejectClassName || 'button--secondary'), onClick: this.onReject },
          this.props.rejectText
        );
      }
      return undefined;
    }
  }]);
  return ConfirmDialog;
}(_react2.default.Component);

ConfirmDialog.propTypes = {
  confirmFunction: _react2.default.PropTypes.func.isRequired,
  confirmText: _react2.default.PropTypes.string,
  confirmClassName: _react2.default.PropTypes.string,
  rejectFunction: _react2.default.PropTypes.func.isRequired,
  rejectText: _react2.default.PropTypes.string,
  rejectClassName: _react2.default.PropTypes.string,
  text: _react2.default.PropTypes.string.isRequired,
  title: _react2.default.PropTypes.string.isRequired,
  extraComponent: _react2.default.PropTypes.object,
  onCloseClick: _react2.default.PropTypes.func,
  onShadowClick: _react2.default.PropTypes.func
};
ConfirmDialog.defaultProps = {
  confirmText: 'Yes',
  rejectText: 'No'
};
exports.default = ConfirmDialog;
