// transcend-react/lib/components/DropdownMultiSelect.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _extends3 = require('babel-runtime/helpers/extends');

var _extends4 = _interopRequireDefault(_extends3);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ReactCSSTransitionGroup = require('react/lib/ReactCSSTransitionGroup');

var _ReactCSSTransitionGroup2 = _interopRequireDefault(_ReactCSSTransitionGroup);

var _DropdownMultiSelectOption = require('./DropdownMultiSelectOption');

var _DropdownMultiSelectOption2 = _interopRequireDefault(_DropdownMultiSelectOption);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropdownMultiSelect = function (_React$Component) {
  (0, _inherits3.default)(DropdownMultiSelect, _React$Component);

  function DropdownMultiSelect(props, context) {
    (0, _classCallCheck3.default)(this, DropdownMultiSelect);

    var _this = (0, _possibleConstructorReturn3.default)(this, (DropdownMultiSelect.__proto__ || (0, _getPrototypeOf2.default)(DropdownMultiSelect)).call(this, props, context));

    _this.state = {
      showOptions: false,
      values: []
    };
    _this.renderDropdownOption = _this.renderDropdownOption.bind(_this);
    _this.renderSelectedText = _this.renderSelectedText.bind(_this);
    _this.onBlur = _this.onBlur.bind(_this);
    _this.onFocus = _this.onFocus.bind(_this);
    _this.onMouseDown = _this.onMouseDown.bind(_this);
    _this.onMouseMove = _this.onMouseMove.bind(_this);
    _this.onMouseLeave = _this.onMouseLeave.bind(_this);
    _this.onKeyDown = _this.onKeyDown.bind(_this);
    _this.onOptionSelect = _this.onOptionSelect.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(DropdownMultiSelect, [{
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.clearTimeout(this.mouseLeaveTimeout);
    }
  }, {
    key: 'onMouseDown',
    value: function onMouseDown() {
      this.setState({
        highlightedIndex: -1
      });
    }
  }, {
    key: 'onOptionSelect',
    value: function onOptionSelect(checked, value) {
      this.setValue(checked, value);
    }
  }, {
    key: 'onMouseMove',
    value: function onMouseMove() {
      if (this.state.showOptions && this.mouseLeaveTimeout) {
        window.clearTimeout(this.mouseLeaveTimeout);
        delete this.mouseLeaveTimeout;
      }
    }
  }, {
    key: 'onMouseLeave',
    value: function onMouseLeave() {
      var _this2 = this;

      if (this.state.showOptions) {
        this.mouseLeaveTimeout = window.setTimeout(function () {
          if (_this2.node) {
            _this2.setState({
              showOptions: false,
              highlightedIndex: -1
            });
          }
        }, 500);
      }
    }
  }, {
    key: 'onFocus',
    value: function onFocus() {
      this.setState({
        showOptions: true,
        highlightedIndex: -1
      });
    }
  }, {
    key: 'onBlur',
    value: function onBlur(evt) {
      if (!evt.relatedTarget) {
        // top component blurred, go ahead and close
        this.setState({
          showOptions: false,
          highlightedIndex: -1
        });
      }
    }
  }, {
    key: 'onKeyDown',
    value: function onKeyDown(evt) {
      if (this.state.showOptions) {
        if (evt.keyCode === 40) {
          // down
          var highlightedIndex = this.state.highlightedIndex + 1;
          if (highlightedIndex >= this.props.options.length) {
            highlightedIndex = 0;
          }
          this.setState({
            highlightedIndex: highlightedIndex
          });
        } else if (evt.keyCode === 38) {
          // up
          var _highlightedIndex = this.state.highlightedIndex - 1;
          if (_highlightedIndex < -1) {
            _highlightedIndex = this.props.options.length - 1;
          }
          this.setState({
            highlightedIndex: _highlightedIndex
          });
        } else if (evt.keyCode === 27) {
          // escape
          this.setState({
            showOptions: false
          });
        } else if (evt.keyCode === 8) {
          // back
          evt.preventDefault();
        } else if (evt.keyCode === 13) {
          if (this.state.highlightedIndex > -1) {
            this.setValue(this.props.options[this.state.highlightedIndex]);
          }
        }
      } else if (evt.keyCode === 38 || evt.keyCode === 40) {
        this.setState({
          showOptions: true,
          highlightedIndex: 0
        });
      }
    }
  }, {
    key: 'setValue',
    value: function setValue(checked, value) {
      if (this.props.valueKey) {
        this.props.onChange(checked, value[this.props.valueKey]);
      } else {
        this.props.onChange(checked, value);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      var values = this.props.values;
      return _react2.default.createElement(
        'div',
        {
          id: this.props.id,
          tabIndex: this.props.tabIndex || 0,
          className: 'dropdown ' + (this.props.className || ''),
          onBlur: this.onBlur,
          onFocus: this.onFocus,
          onMouseDown: this.onMouseDown,
          onMouseMove: this.onMouseMove,
          onMouseLeave: this.onMouseLeave,
          onKeyDown: this.onKeyDown,
          ref: function ref(n) {
            return _this3.node = n;
          }
        },
        _react2.default.createElement(
          'div',
          { className: 'dropdown__container' },
          this.renderSelectedText(values),
          _react2.default.createElement('div', { className: 'dropdown__arrow' })
        ),
        _react2.default.createElement(
          _ReactCSSTransitionGroup2.default,
          { transitionName: 'fadeIn', transitionEnter: true, transitionEnterTimeout: 200, transitionLeaveTimeout: 200 },
          this.renderOptions()
        )
      );
    }
  }, {
    key: 'renderSelectedText',
    value: function renderSelectedText(values) {
      var _this4 = this;

      var lookupObject = this.props.options.reduce(function (map, obj) {
        return (0, _extends4.default)({}, map, (0, _defineProperty3.default)({}, obj[_this4.props.valueKey], obj[_this4.props.textKey]));
      }, {});

      var displayValues = values.map(function (value) {
        return lookupObject[value];
      });

      if (values.length > 0) {
        return _react2.default.createElement(
          'div',
          { className: 'dropdown__selected-text' },
          displayValues.join(', ')
        );
      }
      return _react2.default.createElement(
        'div',
        { className: 'dropdown__selected-text dropdown__selected-text--default' },
        this.props.defaultText
      );
    }
  }, {
    key: 'renderOptions',
    value: function renderOptions() {
      if (this.state.showOptions) {
        var className = ['dropdown__options'];
        if (this.props.alignRight) {
          className.push('dropdown__options--right');
        }

        return _react2.default.createElement(
          'div',
          { className: className.join(' ') },
          this.props.options.map(this.renderDropdownOption)
        );
      }
      return undefined;
    }
  }, {
    key: 'renderDropdownOption',
    value: function renderDropdownOption(option, i) {
      return _react2.default.createElement(_DropdownMultiSelectOption2.default, {
        highlighted: i === this.state.highlightedIndex,
        value: option,
        text: option[this.props.textKey],
        key: option[this.props.textKey],
        onSelect: this.onOptionSelect,
        checkedOptions: this.props.values
      });
    }
  }]);
  return DropdownMultiSelect;
}(_react2.default.Component);

DropdownMultiSelect.propTypes = {
  alignRight: _react2.default.PropTypes.bool,
  className: _react2.default.PropTypes.string,
  defaultText: _react2.default.PropTypes.string,
  id: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func.isRequired,
  options: _react2.default.PropTypes.array.isRequired,
  tabIndex: _react2.default.PropTypes.number,
  textKey: _react2.default.PropTypes.string.isRequired,
  values: _react2.default.PropTypes.any,
  valueKey: _react2.default.PropTypes.string
};
DropdownMultiSelect.defaultProps = {
  textKey: 'text',
  options: [],
  values: []
};
exports.default = DropdownMultiSelect;
