// transcend-react/lib/components/DropdownMultiSelectOption.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropdownMultiSelectOption = function (_Component) {
  (0, _inherits3.default)(DropdownMultiSelectOption, _Component);

  function DropdownMultiSelectOption(props, context) {
    (0, _classCallCheck3.default)(this, DropdownMultiSelectOption);

    var _this = (0, _possibleConstructorReturn3.default)(this, (DropdownMultiSelectOption.__proto__ || (0, _getPrototypeOf2.default)(DropdownMultiSelectOption)).call(this, props, context));

    _this.onChange = _this.onChange.bind(_this);
    _this.isChecked = _this.isChecked.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(DropdownMultiSelectOption, [{
    key: 'onChange',
    value: function onChange(evt) {
      evt.stopPropagation();
      this.props.onSelect(this.checkbox.input.checked, this.props.value);
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var className = 'dropdown__option';
      var classNames = [className];
      if (this.props.highlighted) {
        classNames.push(className + '--highlighted');
      }
      return _react2.default.createElement(
        'div',
        { className: classNames.join(' ') },
        _react2.default.createElement(_Checkbox2.default, {
          value: this.props.text,
          id: this.props.text,
          ref: function ref(n) {
            return _this2.checkbox = n;
          },
          onChange: this.onChange,
          onClick: this.props.onClick,
          checked: this.isChecked(),
          label: this.props.text
        })
      );
    }
  }, {
    key: 'isChecked',
    value: function isChecked() {
      return this.props.checkedOptions.indexOf(this.props.value.text.toLowerCase()) !== -1;
    }
  }]);
  return DropdownMultiSelectOption;
}(_react.Component);

DropdownMultiSelectOption.propTypes = {
  onSelect: _react.PropTypes.func.isRequired,
  highlighted: _react.PropTypes.bool,
  text: _react.PropTypes.string.isRequired,
  value: _react.PropTypes.any.isRequired,
  checkedOptions: _react.PropTypes.array,
  onClick: _react.PropTypes.func
};
exports.default = DropdownMultiSelectOption;
