// transcend-react/lib/components/DropdownSelectOption.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropdownSelectOption = function (_React$Component) {
  (0, _inherits3.default)(DropdownSelectOption, _React$Component);

  function DropdownSelectOption(props, context) {
    (0, _classCallCheck3.default)(this, DropdownSelectOption);

    var _this = (0, _possibleConstructorReturn3.default)(this, (DropdownSelectOption.__proto__ || (0, _getPrototypeOf2.default)(DropdownSelectOption)).call(this, props, context));

    _this.onClick = _this.onClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(DropdownSelectOption, [{
    key: 'onClick',
    value: function onClick(evt) {
      evt.stopPropagation();
      this.props.onSelect(this.props.value);
    }
  }, {
    key: 'render',
    value: function render() {
      var className = 'dropdown__option';
      var classNames = [className];
      if (this.props.highlighted) {
        classNames.push(className + '--highlighted');
      }
      return _react2.default.createElement(
        'div',
        { className: classNames.join(' '), onClick: this.onClick },
        this.props.text
      );
    }
  }]);
  return DropdownSelectOption;
}(_react2.default.Component);

DropdownSelectOption.propTypes = {
  onSelect: _react2.default.PropTypes.func.isRequired,
  highlighted: _react2.default.PropTypes.bool,
  text: _react2.default.PropTypes.string.isRequired,
  value: _react2.default.PropTypes.any.isRequired
};
exports.default = DropdownSelectOption;
