// transcend-react/lib/components/DropdownChecklist.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ComposableDropdown = require('./ComposableDropdown');

var _ComposableDropdown2 = _interopRequireDefault(_ComposableDropdown);

var _DropdownChecklistContent = require('./DropdownChecklistContent');

var _DropdownChecklistContent2 = _interopRequireDefault(_DropdownChecklistContent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropdownChecklist = function (_React$Component) {
  (0, _inherits3.default)(DropdownChecklist, _React$Component);

  function DropdownChecklist() {
    (0, _classCallCheck3.default)(this, DropdownChecklist);
    return (0, _possibleConstructorReturn3.default)(this, (DropdownChecklist.__proto__ || (0, _getPrototypeOf2.default)(DropdownChecklist)).apply(this, arguments));
  }

  (0, _createClass3.default)(DropdownChecklist, [{
    key: 'render',
    value: function render() {
      var selectedText = this.generateSelectedText(this.props.selectedItems);
      var className = this.props.className || '';
      return _react2.default.createElement(
        _ComposableDropdown2.default,
        { defaultText: this.props.defaultText, selectedText: selectedText, className: 'dropdown-checklist ' + className },
        _react2.default.createElement(_DropdownChecklistContent2.default, {
          items: this.props.items,
          onChange: this.props.onChange,
          selectedItems: this.props.selectedItems || [],
          textKey: this.props.textKey,
          valueKey: this.props.valueKey,
          imageKey: this.props.imageKey
        })
      );
    }
  }, {
    key: 'generateSelectedText',
    value: function generateSelectedText(selectedItems) {
      var _this2 = this;

      var selectedText = '';
      if (selectedItems.length > 0) {
        var displayValues = selectedItems.map(function (item) {
          if (_this2.props.generateSelectedText) {
            return _this2.props.generateSelectedText(item);
          }
          return item[_this2.props.textKey];
        }, this);
        selectedText = displayValues.join(', ');
      }
      return selectedText;
    }
  }]);
  return DropdownChecklist;
}(_react2.default.Component);

DropdownChecklist.propTypes = {
  className: _react2.default.PropTypes.string,
  items: _react2.default.PropTypes.array.isRequired,
  selectedItems: _react2.default.PropTypes.array,
  onChange: _react2.default.PropTypes.func.isRequired,
  textKey: _react2.default.PropTypes.string.isRequired,
  imageKey: _react2.default.PropTypes.string,
  valueKey: _react2.default.PropTypes.string,
  generateSelectedText: _react2.default.PropTypes.func,
  defaultText: _react2.default.PropTypes.string
};
DropdownChecklist.defaultProps = {
  textKey: 'text',
  items: [],
  selectedItems: []
};
exports.default = DropdownChecklist;
