// transcend-react/lib/components/DropdownChecklistContentItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropdownChecklistContentItem = function (_React$Component) {
  (0, _inherits3.default)(DropdownChecklistContentItem, _React$Component);

  function DropdownChecklistContentItem() {
    var _ref;

    (0, _classCallCheck3.default)(this, DropdownChecklistContentItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = DropdownChecklistContentItem.__proto__ || (0, _getPrototypeOf2.default)(DropdownChecklistContentItem)).call.apply(_ref, [this].concat(args)));

    _this.onClick = _this.onClick.bind(_this);
    _this.onCheckChange = _this.onCheckChange.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(DropdownChecklistContentItem, [{
    key: 'onClick',
    value: function onClick() {
      if (this.props.onChange) {
        this.props.onChange(!this.props.selected, this.props.item);
      }
    }
  }, {
    key: 'onCheckChange',
    value: function onCheckChange() {
      if (this.props.onChange) {
        this.props.onChange(this.checkbox.input.checked, this.props.item);
      }
    }

    /* eslint-disable jsx-a11y/no-static-element-interactions */

  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var item = this.props.item || {};
      var displayName = this.props.item[this.props.textKey] || '';
      var selectedClass = '';
      if (this.props.selected) {
        selectedClass = 'dropdown-checklist__item--selected';
      }
      return _react2.default.createElement(
        'div',
        { key: item[this.props.valueKey], className: 'dropdown-checklist__item ' + selectedClass },
        _react2.default.createElement(
          'div',
          { className: 'dropdown-checklist__item__info', onClick: this.onClick },
          _react2.default.createElement('img', { src: item[this.props.imageKey], alt: item[this.props.textKey] }),
          _react2.default.createElement(
            'span',
            { className: 'dropdown-checklist__item__name' },
            displayName
          )
        ),
        _react2.default.createElement(_Checkbox2.default, {
          id: item[this.props.valueKey],
          className: 'dropdown-checklist',
          checked: this.props.selected,
          value: item[this.props.valueKey],
          onChange: this.onCheckChange,
          ref: function ref(n) {
            return _this2.checkbox = n;
          }
        })
      );
    }
  }]);
  return DropdownChecklistContentItem;
}(_react2.default.Component);

DropdownChecklistContentItem.propTypes = {
  className: _react2.default.PropTypes.string,
  item: _react2.default.PropTypes.object,
  textKey: _react2.default.PropTypes.string,
  valueKey: _react2.default.PropTypes.string,
  imageKey: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func,
  selected: _react2.default.PropTypes.bool
};
exports.default = DropdownChecklistContentItem;
