// transcend-react/lib/components/FlashNotification.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ReactCSSTransitionGroup = require('react/lib/ReactCSSTransitionGroup');

var _ReactCSSTransitionGroup2 = _interopRequireDefault(_ReactCSSTransitionGroup);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FlashNotification = function (_React$Component) {
  (0, _inherits3.default)(FlashNotification, _React$Component);

  function FlashNotification() {
    var _ref;

    (0, _classCallCheck3.default)(this, FlashNotification);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = FlashNotification.__proto__ || (0, _getPrototypeOf2.default)(FlashNotification)).call.apply(_ref, [this].concat(args)));

    _this.state = {};
    _this.onStart = _this.onStart.bind(_this);
    _this.onComplete = _this.onComplete.bind(_this);
    _this.onHideClick = _this.onHideClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(FlashNotification, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.timeout = window.setTimeout(this.onStart);
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(props) {
      if (!_lodash2.default.isEqual(this.props, props)) {
        window.clearTimeout(this.timeout);
        this.timeout = window.setTimeout(this.onStart);
      }
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.clearTimeout(this.timeout);
    }
  }, {
    key: 'onStart',
    value: function onStart() {
      var _this2 = this;

      this.setState({
        show: true
      }, function () {
        _this2.timeout = window.setTimeout(_this2.onComplete, _this2.props.duration);
      });
    }
  }, {
    key: 'onComplete',
    value: function onComplete() {
      var _this3 = this;

      window.clearTimeout(this.timeout);
      this.setState({
        show: false
      }, function () {
        window.clearTimeout(_this3.timeout);
        if (_this3.props.handleClose) {
          _this3.props.handleClose();
        }
        if (_this3.props.onClose) {
          _this3.props.onClose();
        }
      });
    }
  }, {
    key: 'onHideClick',
    value: function onHideClick() {
      this.onComplete();
    }
  }, {
    key: 'getMessages',
    value: function getMessages() {
      if (this.props.message) {
        return [this.props.message];
      }
      return this.props.flash.map(function (flash) {
        if (_lodash2.default.isArray(flash) && flash.length > 1) {
          return flash[1];
        }
        return flash;
      });
    }
  }, {
    key: 'getType',
    value: function getType() {
      return this.props.type || this.getDeprecatedType();
    }
  }, {
    key: 'getDeprecatedType',
    value: function getDeprecatedType() {
      var type = 'info';
      if (_lodash2.default.isArray(this.props.flash) && this.props.flash.length) {
        var first = this.props.flash[0];
        if (first && first.length > 1) {
          type = first[0];
        }
      }
      return type;
    }
  }, {
    key: 'render',
    value: function render() {
      var style = {};
      if (!this.state.show) {
        style.display = 'none';
      }

      return _react2.default.createElement(
        'div',
        { className: 'flash__container', style: style },
        _react2.default.createElement(
          _ReactCSSTransitionGroup2.default,
          {
            transitionName: 'fadeIn',
            transitionEnter: true,
            transitionEnterTimeout: 200,
            transitionLeave: true,
            transitionLeaveTimeout: 200
          },
          this.renderFlash()
        )
      );
    }
  }, {
    key: 'renderFlash',
    value: function renderFlash() {
      if (this.state.show) {
        var type = this.getType();
        var messages = this.getMessages();
        var className = ['flash', 'flash--' + type];
        return _react2.default.createElement(
          'div',
          { className: className.join(' ') },
          this.props.children || messages.map(this.renderMessage),
          _react2.default.createElement(
            'div',
            { className: 'flash__close', onClick: this.onHideClick },
            _react2.default.createElement('i', { className: 'icon icon--close' })
          )
        );
      }
      return undefined;
    }
  }, {
    key: 'renderMessage',
    value: function renderMessage(message, i) {
      return _react2.default.createElement(
        'div',
        { key: i, className: 'flash__message' },
        message
      );
    }
  }]);
  return FlashNotification;
}(_react2.default.Component);

FlashNotification.propTypes = {
  flash: _react2.default.PropTypes.array,
  isDisplayed: _react2.default.PropTypes.bool,
  handleClose: _react2.default.PropTypes.func,
  onClose: _react2.default.PropTypes.func,
  duration: _react2.default.PropTypes.number,
  type: _react2.default.PropTypes.string,
  message: _react2.default.PropTypes.string,
  children: _react2.default.PropTypes.node
};
FlashNotification.defaultProps = {
  duration: 5000
};
exports.default = FlashNotification;
