// transcend-react/lib/components/Image.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Loader = require('./Loader');

var _Loader2 = _interopRequireDefault(_Loader);

var _fallback = require('../../stylesheets/images/fallback.svg');

var _fallback2 = _interopRequireDefault(_fallback);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var defaultImage = _fallback2.default;

var Image = function (_Component) {
  (0, _inherits3.default)(Image, _Component);

  function Image(props) {
    var _ref;

    (0, _classCallCheck3.default)(this, Image);

    for (var _len = arguments.length, args = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = Image.__proto__ || (0, _getPrototypeOf2.default)(Image)).call.apply(_ref, [this, props].concat(args)));

    _initialiseProps.call(_this);

    _this.state = {
      isLoading: true
    };
    return _this;
  }

  (0, _createClass3.default)(Image, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(newProps) {
      if (newProps.src !== this.props.src) {
        this.setState({
          hasError: false,
          isLoading: true
        });
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'span',
        { className: 'image' },
        this.state.isLoading ? _react2.default.createElement(_Loader2.default, null) : undefined,
        _react2.default.createElement('img', {
          alt: this.props.alt,
          crossOrigin: this.props.crossOrigin,
          height: this.props.height,
          sizes: this.props.sizes,
          src: this.state.hasError ? this.props.fallback : this.props.src || this.props.fallback,
          srcSet: this.props.srcSet,
          width: this.props.width,
          onError: this.state.hasError ? undefined : this.onError,
          onLoad: this.state.hasError ? undefined : this.onLoad,
          className: this.props.className
        })
      );
    }
  }]);
  return Image;
}(_react.Component);

Image.propTypes = {
  alt: _react.PropTypes.string,
  crossOrigin: _react.PropTypes.string,
  fallback: _react.PropTypes.string,
  height: _react.PropTypes.string,
  sizes: _react.PropTypes.string,
  src: _react.PropTypes.string,
  srcSet: _react.PropTypes.string,
  width: _react.PropTypes.string,
  onLoad: _react.PropTypes.func,
  onError: _react.PropTypes.func,
  className: _react.PropTypes.string
};
Image.defaultProps = {
  fallback: defaultImage
};

var _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.onLoad = function () {
    for (var _len2 = arguments.length, args = Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
      args[_key2] = arguments[_key2];
    }

    _this2.setState({
      isLoading: false
    }, function () {
      if (_this2.props.onLoad) {
        var _props;

        (_props = _this2.props).onLoad.apply(_props, args);
      }
    });
  };

  this.onError = function () {
    for (var _len3 = arguments.length, args = Array(_len3), _key3 = 0; _key3 < _len3; _key3++) {
      args[_key3] = arguments[_key3];
    }

    _this2.setState({
      hasError: true,
      isLoading: false
    }, function () {
      if (_this2.props.onError) {
        var _props2;

        (_props2 = _this2.props).onError.apply(_props2, args);
      }
    });
  };
};

exports.default = Image;
