// transcend-react/lib/components/Loader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Loader;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var children = _lodash2.default.times(36).map(function (i) {
  return _react2.default.createElement(
    'circle',
    { key: i, r: '4', fill: 'white', opacity: '0' },
    _react2.default.createElement(
      'animateMotion',
      { begin: i * (1 / 45), dur: '2s', repeatCount: 'indefinite' },
      _react2.default.createElement('mpath', { xlinkHref: '#loaderMotion' })
    ),
    _react2.default.createElement('animate', {
      begin: i * (1 / 45),
      attributeType: 'XML',
      attributeName: 'opacity',
      from: '0',
      to: '1',
      dur: '1s',
      fill: 'freeze',
      restart: 'never'
    })
  );
});
function Loader() {
  /* eslint-disable max-len */
  return _react2.default.createElement(
    'div',
    { className: 'loader' },
    _react2.default.createElement(
      'svg',
      {
        width: '72',
        height: '34',
        xmlns: 'http://www.w3.org/2000/svg',
        version: '1.1',
        xmlnsXlink: 'http://www.w3.org/1999/xlink',
        ref: function ref(n) {
          return n && n.setAttribute('timelineBegin', 'onStart');
        }
      },
      _react2.default.createElement('path', { id: 'loaderMotion', d: 'M67.3,17.8c0-9.3-4.8-14.5-13.8-14.5C41.9,3.2,35.7,17,35.7,17s-6.2,13.8-17.8,13.8C10,30.8,4,25.5,4,17.7 C4, 9.6, 10, 3.2, 17.8, 3.2C29.4, 3.2, 35.7, 17, 35.7, 17s6.2, 13.8, 17.8, 13.8C61.4, 30.8, 67.3, 24.8, 67.3, 17.8z' }),
      children
    )
  );
}
