// transcend-react/lib/components/Modal.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Modal = function (_React$Component) {
  (0, _inherits3.default)(Modal, _React$Component);
  (0, _createClass3.default)(Modal, null, [{
    key: 'defaultProps',
    value: function defaultProps() {
      return {
        closeButton: true
      };
    }
  }]);

  function Modal(props, context) {
    (0, _classCallCheck3.default)(this, Modal);

    var _this = (0, _possibleConstructorReturn3.default)(this, (Modal.__proto__ || (0, _getPrototypeOf2.default)(Modal)).call(this, props, context));

    _this.state = {
      isDisplayed: false
    };
    _this.onModalShadowClick = _this.onModalShadowClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(Modal, [{
    key: 'onModalShadowClick',
    value: function onModalShadowClick(evt) {
      // only exactly it
      if (evt.target === this.shadow || evt.target === this.fill) {
        if (this.props.onShadowClick) {
          this.props.onShadowClick(evt);
        }
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        'div',
        { className: 'modal ' + (this.props.className || '') },
        _react2.default.createElement('div', { className: 'modal__shadow', onClick: this.onModalShadowClick, ref: function ref(n) {
            return _this2.shadow = n;
          } }),
        _react2.default.createElement(
          'div',
          { className: 'modal__fill', onClick: this.onModalShadowClick, ref: function ref(n) {
              return _this2.fill = n;
            } },
          _react2.default.createElement(
            'div',
            { className: 'modal__content' },
            this.props.children,
            this.renderCloseButton()
          )
        )
      );
    }
  }, {
    key: 'renderCloseButton',
    value: function renderCloseButton() {
      if (this.props.onCloseClick) {
        return _react2.default.createElement(
          'div',
          { className: 'modal__close', onClick: this.props.onCloseClick },
          _react2.default.createElement('i', { className: 'icon icon--close' })
        );
      }
      return undefined;
    }
  }]);
  return Modal;
}(_react2.default.Component);

Modal.propTypes = {
  children: _react2.default.PropTypes.node,
  className: _react2.default.PropTypes.string,
  onCloseClick: _react2.default.PropTypes.func,
  onShadowClick: _react2.default.PropTypes.func
};
exports.default = Modal;
