// transcend-react/lib/components/MultiSelectItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MultiSelectItem = function (_React$Component) {
  (0, _inherits3.default)(MultiSelectItem, _React$Component);

  function MultiSelectItem() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, MultiSelectItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = MultiSelectItem.__proto__ || (0, _getPrototypeOf2.default)(MultiSelectItem)).call.apply(_ref, [this].concat(args))), _this), _this.onChange = function () {
      if (_this.props.onChange) {
        _this.props.onChange(_this.props.item);
      }
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(MultiSelectItem, [{
    key: 'render',
    value: function render() {
      var id = this.props.groupName + '_' + this.props.item.value;

      return _react2.default.createElement(
        'div',
        { className: 'multi-select__items__item' },
        _react2.default.createElement(_Checkbox2.default, {
          id: id,
          checked: this.props.item.checked,
          onChange: this.onChange,
          label: this.props.item.text
        })
      );
    }
  }]);
  return MultiSelectItem;
}(_react2.default.Component);

MultiSelectItem.propTypes = {
  groupName: _react2.default.PropTypes.string.isRequired,
  item: _react2.default.PropTypes.object.isRequired,
  onChange: _react2.default.PropTypes.func
};
exports.default = MultiSelectItem;
