// transcend-react/lib/components/NavDropdownSelectOption.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NavDropdownSelectOption = function (_React$Component) {
  (0, _inherits3.default)(NavDropdownSelectOption, _React$Component);

  function NavDropdownSelectOption(props, context) {
    (0, _classCallCheck3.default)(this, NavDropdownSelectOption);

    var _this = (0, _possibleConstructorReturn3.default)(this, (NavDropdownSelectOption.__proto__ || (0, _getPrototypeOf2.default)(NavDropdownSelectOption)).call(this, props, context));

    _this.onClick = _this.onClick.bind(_this);
    _this.onIconClick = _this.onIconClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(NavDropdownSelectOption, [{
    key: 'onClick',
    value: function onClick() {
      if (this.props.onClick) {
        this.props.onClick(this.props.option);
      }
    }
  }, {
    key: 'onIconClick',
    value: function onIconClick(evt) {
      if (this.props.onIconClick) {
        evt.preventDefault();
        evt.stopPropagation();
        this.props.onIconClick(this.props.option);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var icon = this.props.icon || this.props.option.icon;
      var iconClassName = '';
      if (icon) {
        iconClassName = 'dropdown__option--has-icon';
      }
      return _react2.default.createElement(
        'div',
        { className: 'dropdown__option ' + iconClassName, onClick: this.onClick },
        this.props.option[this.props.textKey],
        this.renderIcon()
      );
    }
  }, {
    key: 'renderIcon',
    value: function renderIcon() {
      var icon = this.props.icon || this.props.option.icon;
      if (icon) {
        return _react2.default.createElement('i', { className: 'icon icon--' + icon + ' select__options__option__icon', onClick: this.onIconClick });
      }
      return undefined;
    }
  }]);
  return NavDropdownSelectOption;
}(_react2.default.Component);

NavDropdownSelectOption.propTypes = {
  onClick: _react2.default.PropTypes.func.isRequired,
  onIconClick: _react2.default.PropTypes.func,
  textKey: _react2.default.PropTypes.string.isRequired,
  option: _react2.default.PropTypes.any.isRequired,
  icon: _react2.default.PropTypes.string
};
exports.default = NavDropdownSelectOption;
