// transcend-react/lib/components/NavLink.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = NavLink;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function locationMatches(location, match) {
  if (typeof match === 'string') {
    return location === match;
  }
  return !!location.match(match);
}

function NavLink(props, context) {
  var location = context.router.createLocation(props.location.pathname);
  var className = [props.className];
  if (locationMatches('' + (location.basename || '') + location.pathname, props.match || props.to)) {
    className.push(props.matchedClassName);
  }
  if (props.absolute) {
    return _react2.default.createElement(
      'a',
      { href: props.to, className: className.join(' ') },
      props.children
    );
  }
  return _react2.default.createElement(
    _reactRouter.Link,
    { to: props.to, className: className.join(' ') },
    props.children
  );
}

NavLink.contextTypes = {
  router: _react.PropTypes.object
};

NavLink.propTypes = {
  absolute: _react.PropTypes.bool,
  children: _react.PropTypes.node,
  className: _react.PropTypes.string,
  location: _react.PropTypes.object.isRequired,
  match: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object]),
  matchedClassName: _react.PropTypes.string.isRequired,
  to: _react.PropTypes.string.isRequired
};
