// transcend-react/lib/components/NavTabs.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NavTabs = function (_React$Component) {
  (0, _inherits3.default)(NavTabs, _React$Component);

  function NavTabs() {
    (0, _classCallCheck3.default)(this, NavTabs);
    return (0, _possibleConstructorReturn3.default)(this, (NavTabs.__proto__ || (0, _getPrototypeOf2.default)(NavTabs)).apply(this, arguments));
  }

  (0, _createClass3.default)(NavTabs, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { className: 'nav-tabs ' + (this.props.className || '') },
        _react2.default.createElement(
          'ul',
          { className: 'nav-tabs__list' },
          this.renderChildren()
        )
      );
    }
  }, {
    key: 'renderChildren',
    value: function renderChildren() {
      var _this2 = this;

      if (this.props.children) {
        return _react2.default.Children.map(this.props.children, function (child, i) {
          var className = ['nav-tabs__tab'];
          var to = child.props.to || child.props.href;
          if (_this2.shouldMarkAsSelected(to)) {
            className.push('nav-tabs__tab--selected');
          }
          return _react2.default.createElement(
            'li',
            { key: i, className: className.join(' ') },
            child
          );
        });
      }
      return this.renderLinks();
    }
  }, {
    key: 'renderLinks',
    value: function renderLinks() {
      var _this3 = this;

      return this.props.links.map(function (link, i) {
        var className = ['nav-tabs__tab'];
        if (_this3.shouldMarkAsSelected(link.url)) {
          className.push('nav-tabs__tab--selected');
        }
        if (link.className) {
          className.push(link.className);
        }
        return _react2.default.createElement(
          'li',
          { key: i, className: className.join(' ') },
          _react2.default.createElement(
            _reactRouter.Link,
            { to: link.url },
            link.name
          )
        );
      });
    }
  }, {
    key: 'shouldMarkAsSelected',
    value: function shouldMarkAsSelected(url) {
      var pathname = this.props.location.pathname || '';
      if (this.props.fuzzy) {
        return pathname.indexOf(url) !== -1;
      } else {
        return this.props.location.pathname === url;
      }
    }
  }]);
  return NavTabs;
}(_react2.default.Component);

NavTabs.propTypes = {
  links: _react2.default.PropTypes.array,
  className: _react2.default.PropTypes.string,
  location: _react2.default.PropTypes.object,
  children: _react2.default.PropTypes.node,
  fuzzy: _react2.default.PropTypes.bool
};
NavTabs.defaultProps = {
  location: {}
};
exports.default = NavTabs;
