// transcend-react/lib/components/PhoneInput.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _KeyCode = require('../helpers/KeyCode');

var _KeyCode2 = _interopRequireDefault(_KeyCode);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var PhoneInput = function (_Component) {
  (0, _inherits3.default)(PhoneInput, _Component);

  function PhoneInput(props) {
    var _ref;

    (0, _classCallCheck3.default)(this, PhoneInput);

    for (var _len = arguments.length, args = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = PhoneInput.__proto__ || (0, _getPrototypeOf2.default)(PhoneInput)).call.apply(_ref, [this, props].concat(args)));

    var phone = '';
    if (props.phone) {
      phone = _this.displayPhone(props.phone);
    }

    _this.state = {
      phone: phone
    };

    _this.onChange = _this.onChange.bind(_this);
    _this.onFocus = _this.onFocus.bind(_this);
    _this.onBlur = _this.onBlur.bind(_this);
    _this.getValue = _this.getValue.bind(_this);
    _this.valueIsValid = _this.valueIsValid.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(PhoneInput, [{
    key: 'onKeyDown',
    value: function onKeyDown(event) {
      if (_KeyCode2.default.isEnterKey(event)) {
        return true;
      } else if (!_KeyCode2.default.isValidPhoneKey(event)) {
        event.preventDefault();
        event.stopPropagation();
      }
      return false;
    }
  }, {
    key: 'onChange',
    value: function onChange(event) {
      var phone = event.target.value;
      this.setState({
        phone: phone
      }, this.props.onChange(phone));
    }
  }, {
    key: 'onFocus',
    value: function onFocus(event) {
      this.props.onFocus(event);
    }
  }, {
    key: 'onBlur',
    value: function onBlur(event) {
      this.props.onBlur(event);
    }
  }, {
    key: 'getValue',
    value: function getValue() {
      var numericPhone = this.state.phone.replace(/[^\d]/g, '');
      return numericPhone;
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement('input', {
        type: 'text',
        placeholder: '1-555-867-5309',
        onChange: this.onChange,
        onKeyDown: this.onKeyDown,
        onFocus: this.onFocus,
        onBlur: this.onBlur,
        value: this.state.phone
      });
    }
  }, {
    key: 'valueIsValid',
    value: function valueIsValid() {
      if (!this.state.phone) {
        return false;
      }
      var numericPhone = this.state.phone.replace(/[^\d]/g, '');

      if (numericPhone.length !== 11) {
        return false;
      }
      return true;
    }
  }, {
    key: 'displayPhone',
    value: function displayPhone(phone) {
      if (phone) {
        return phone.slice(0, 1) + '-' + phone.slice(1, 4) + '-' + phone.slice(4, 7) + '-' + phone.slice(7, 11);
      } else {
        return 'No Phone #';
      }
    }
  }]);
  return PhoneInput;
}(_react.Component);

PhoneInput.propTypes = {
  onChange: _react.PropTypes.func.isRequired,
  onFocus: _react.PropTypes.func.isRequired,
  onBlur: _react.PropTypes.func.isRequired,
  phone: _react.PropTypes.string
};
exports.default = PhoneInput;
