// transcend-react/lib/components/Radio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require("babel-runtime/helpers/extends");

var _extends3 = _interopRequireDefault(_extends2);

exports.default = Radio;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Radio(props) {
  var inputProps = (0, _extends3.default)({}, props);
  delete inputProps.label;

  return _react2.default.createElement(
    "div",
    null,
    _react2.default.createElement("input", (0, _extends3.default)({
      type: "radio"
    }, inputProps)),
    _react2.default.createElement(
      "label",
      { htmlFor: props.id },
      props.label
    )
  );
}

Radio.propTypes = {
  id: _react2.default.PropTypes.string.isRequired,
  label: _react2.default.PropTypes.string.isRequired,
  value: _react2.default.PropTypes.string.isRequired,
  disabled: _react2.default.PropTypes.bool,
  name: _react2.default.PropTypes.string,
  readOnly: _react2.default.PropTypes.bool,
  onChange: _react2.default.PropTypes.func
};
