// transcend-react/lib/components/SelectorPanelItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SelectorPanelItem = function (_React$Component) {
  (0, _inherits3.default)(SelectorPanelItem, _React$Component);

  function SelectorPanelItem() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, SelectorPanelItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = SelectorPanelItem.__proto__ || (0, _getPrototypeOf2.default)(SelectorPanelItem)).call.apply(_ref, [this].concat(args))), _this), _this.onClick = function (evt) {
      evt.preventDefault();
      evt.stopPropagation();
      _this.props.onSelect(_this.props.item);
    }, _this.onOptionsClick = function (evt) {
      evt.preventDefault();
      evt.stopPropagation();
      _this.props.onOptions(_this.props.item);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(SelectorPanelItem, [{
    key: 'render',
    value: function render() {
      /* eslint-disable  jsx-a11y/no-static-element-interactions */
      return _react2.default.createElement(
        'li',
        { onClick: this.onClick, className: this.props.className || '' },
        _react2.default.createElement(
          'div',
          { className: 'selector-panel__item__container' },
          _react2.default.createElement(
            'div',
            { className: 'selector-panel__item__name' },
            this.props.item[this.props.nameAttribute]
          )
        )
      );
    }
  }]);
  return SelectorPanelItem;
}(_react2.default.Component);

SelectorPanelItem.propTypes = {
  item: _react2.default.PropTypes.object.isRequired,
  onSelect: _react2.default.PropTypes.func.isRequired,
  onOptions: _react2.default.PropTypes.func.isRequired,
  className: _react2.default.PropTypes.string,
  nameAttribute: _react2.default.PropTypes.string.isRequired
};
exports.default = SelectorPanelItem;
