// transcend-react/lib/components/Slider.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Slider;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Slider(_ref) {
  var layoutType = _ref.layoutType,
      id = _ref.id,
      checked = _ref.checked,
      value = _ref.value,
      onChange = _ref.onChange,
      labelTextChecked = _ref.labelTextChecked,
      labelTextUncheked = _ref.labelTextUncheked;

  var extraClassName = layoutType === 'inline' ? 'inline' : '';
  return _react2.default.createElement(
    'div',
    { className: 'slider ' + extraClassName },
    _react2.default.createElement('input', {
      type: 'checkbox',
      id: id,
      name: id,
      checked: checked,
      defaultValue: value,
      onChange: onChange
    }),
    _react2.default.createElement('label', { htmlFor: id }),
    labelTextChecked && labelTextUncheked && _react2.default.createElement(
      'span',
      { className: 'label-text' },
      checked ? labelTextChecked : labelTextUncheked
    )
  );
}

Slider.defaultProps = {
  checked: false
};

Slider.propTypes = {
  checked: _react2.default.PropTypes.bool,
  id: _react2.default.PropTypes.string.isRequired,
  layoutType: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func.isRequired,
  value: _react2.default.PropTypes.string.isRequired,
  labelTextChecked: _react2.default.PropTypes.string,
  labelTextUncheked: _react2.default.PropTypes.string
};
