// transcend-react/lib/components/SliderInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require("babel-runtime/core-js/object/get-prototype-of");

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require("babel-runtime/helpers/createClass");

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SliderInput = function (_React$Component) {
  (0, _inherits3.default)(SliderInput, _React$Component);

  function SliderInput() {
    var _ref;

    (0, _classCallCheck3.default)(this, SliderInput);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = SliderInput.__proto__ || (0, _getPrototypeOf2.default)(SliderInput)).call.apply(_ref, [this].concat(args)));

    _this.onInputChange = _this.onInputChange.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(SliderInput, [{
    key: "onChange",
    value: function onChange(value) {
      var newValue = value * 1;
      if (newValue > this.props.max) {
        newValue = this.props.max;
      } else if (newValue < this.props.min) {
        newValue = this.props.min;
      }
      this.props.onChange(newValue);
    }
  }, {
    key: "onInputChange",
    value: function onInputChange(evt) {
      this.onChange(evt.target.value);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        "div",
        { className: "slider-input" },
        _react2.default.createElement(
          "div",
          { className: "flex" },
          _react2.default.createElement(
            "div",
            { className: "shrink slider-input__text slider-input__text--min" },
            this.props.minText
          ),
          _react2.default.createElement(
            "div",
            { className: " slider-input__inputs grow" },
            _react2.default.createElement("input", {
              className: "slider-input__range",
              ref: function ref(n) {
                return _this2.slider = n;
              },
              type: "range",
              min: this.props.min,
              max: this.props.max,
              step: this.props.step,
              value: this.props.value,
              onChange: this.onInputChange
            }),
            _react2.default.createElement("input", {
              ref: function ref(n) {
                return _this2.sliderInput = n;
              },
              type: "number",
              min: this.props.min,
              max: this.props.max,
              step: this.props.step,
              className: "slider-input__manual",
              value: this.props.value,
              onChange: this.onInputChange
            })
          ),
          _react2.default.createElement(
            "div",
            { className: "shrink slider-input__text slider-input__text--max" },
            this.props.maxText
          )
        )
      );
    }
  }]);
  return SliderInput;
}(_react2.default.Component);

SliderInput.propTypes = {
  min: _react2.default.PropTypes.number.isRequired,
  max: _react2.default.PropTypes.number.isRequired,
  maxText: _react2.default.PropTypes.string,
  minText: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func,
  step: _react2.default.PropTypes.number,
  value: _react2.default.PropTypes.number
};
SliderInput.defaultProps = {
  step: 0,
  min: 0
};
exports.default = SliderInput;
