// transcend-react/lib/components/UserMenu.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ReactCSSTransitionGroup = require('react/lib/ReactCSSTransitionGroup');

var _ReactCSSTransitionGroup2 = _interopRequireDefault(_ReactCSSTransitionGroup);

var _Navigation = require('../Navigation');

var _Navigation2 = _interopRequireDefault(_Navigation);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function renderLink(link, i) {
  return _react2.default.createElement(
    'li',
    { className: 'user-menu__link', key: i },
    _react2.default.createElement(
      'a',
      { href: link.href },
      link.name
    )
  );
}

var UserMenu = function (_Component) {
  (0, _inherits3.default)(UserMenu, _Component);

  function UserMenu() {
    var _ref;

    (0, _classCallCheck3.default)(this, UserMenu);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = (0, _possibleConstructorReturn3.default)(this, (_ref = UserMenu.__proto__ || (0, _getPrototypeOf2.default)(UserMenu)).call.apply(_ref, [this].concat(args)));

    _this.state = {};
    _this.onShowClick = _this.onShowClick.bind(_this);
    _this.onLogoutClick = _this.onLogoutClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(UserMenu, [{
    key: 'onShowClick',
    value: function onShowClick(evt) {
      evt.preventDefault();
      this.setState({
        showMenu: !this.state.showMenu
      });
    }
  }, {
    key: 'onLogoutClick',
    value: function onLogoutClick(evt) {
      evt.preventDefault();
      var form = window.document.createElement('FORM');
      form.action = this.props.logoutUrl || _Navigation2.default.buildPath('/logout');
      form.method = 'POST';
      window.document.body.appendChild(form);
      var methodInput = window.document.createElement('input');
      methodInput.value = 'POST';
      methodInput.name = '_method';
      methodInput.type = 'hidden';
      form.appendChild(methodInput);
      var csrfInput = window.document.createElement('input');
      csrfInput.value = this.props.csrfToken;
      csrfInput.name = 'authenticity_token';
      methodInput.type = 'hidden';
      form.appendChild(methodInput);
      form.submit();
    }
  }, {
    key: 'getName',
    value: function getName() {
      if (this.props.user.first_name) {
        return this.props.user.first_name + ' ' + this.props.user.last_name;
      }
      return this.props.user.username;
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { className: 'user-menu' },
        _react2.default.createElement(
          'a',
          { className: 'user-menu__user', onClick: this.onShowClick, tabIndex: '0' },
          this.renderUser()
        ),
        _react2.default.createElement(
          _ReactCSSTransitionGroup2.default,
          {
            transitionName: 'fadeIn',
            transitionEnter: true,
            transitionEnterTimeout: 200,
            transitionLeaveTimeout: 200
          },
          this.renderMenu()
        )
      );
    }
  }, {
    key: 'renderMenu',
    value: function renderMenu() {
      var links = [{ name: 'My Settings', href: '/settings/personal_settings' }, { name: 'Social Profiles', href: '/settings/social_profiles' }];
      if (this.props.showSupport) {
        links.push({ name: 'Support', href: 'https://support.nuvi.com/support/tickets/new' });
      }
      if (this.state.showMenu) {
        return _react2.default.createElement(
          'div',
          { className: 'user-menu__container' },
          _react2.default.createElement('div', { className: 'user-menu__scrim', onClick: this.onShowClick }),
          _react2.default.createElement(
            'div',
            { className: 'user-menu__menu' },
            _react2.default.createElement(
              'ul',
              null,
              links.map(renderLink)
            ),
            _react2.default.createElement(
              'div',
              { className: 'user-menu__logout' },
              _react2.default.createElement(
                'a',
                { href: this.props.logoutUrl || _Navigation2.default.buildPath('/logout'), onClick: this.onLogoutClick },
                'Logout'
              )
            ),
            _react2.default.createElement(
              'div',
              { className: 'user-menu__user' },
              this.renderUser()
            )
          )
        );
      }
      return undefined;
    }
  }, {
    key: 'renderUser',
    value: function renderUser() {
      return [_react2.default.createElement('img', {
        key: 'image',
        src: this.props.user.avatar,
        role: 'presentation',
        alt: this.props.user.username,
        className: 'user-menu__user__image'
      }), _react2.default.createElement(
        'span',
        {
          key: 'name',
          className: 'user-menu__user__name'
        },
        this.getName()
      )];
    }
  }]);
  return UserMenu;
}(_react.Component);

UserMenu.propTypes = {
  user: _react.PropTypes.object.isRequired,
  showSupport: _react.PropTypes.bool,
  logoutUrl: _react.PropTypes.string,
  csrfToken: _react.PropTypes.string
};
exports.default = UserMenu;
