// transcend-react/lib/helpers/KeyCode.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var KeyCode = {
  isValidClockKey: function isValidClockKey(event) {
    return KeyCode.isNumericKey(event) || KeyCode.isNavigationKey(event);
  },
  isValidPhoneKey: function isValidPhoneKey(event) {
    return KeyCode.isNumericKey(event) || KeyCode.isOtherPhoneKey(event) || KeyCode.isNavigationKey(event);
  },
  isOtherPhoneKey: function isOtherPhoneKey(event) {
    // - -(number pad)
    var keyCode = event.keyCode;
    return [189, 109].indexOf(keyCode) >= 0 && !event.shiftKey;
  },
  isNumericKey: function isNumericKey(event) {
    // 0-9, number pad 0-9
    var keyCode = event.keyCode;
    return [48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105].indexOf(keyCode) >= 0 && !event.shiftKey;
  },
  isNavigationKey: function isNavigationKey(event) {
    // backspace, tab, left, right, delete
    var keyCode = event.keyCode;
    return [8, 9, 37, 39, 46].indexOf(keyCode) >= 0;
  },
  isAlphabetKey: function isAlphabetKey(event) {
    // a-z
    var keyCode = event.keyCode;
    return [65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90].indexOf(keyCode) >= 0;
  },
  isUpArrow: function isUpArrow(event) {
    return event.keyCode === 38;
  },
  isDownArrow: function isDownArrow(event) {
    return event.keyCode === 40;
  },
  isEnterKey: function isEnterKey(event) {
    return event.keyCode === 13;
  }
};
exports.default = KeyCode;
