// transcend-react/lib/helpers/Themes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _ColorPalette = require('../ColorPalette');

var _ColorPalette2 = _interopRequireDefault(_ColorPalette);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {

  graphite: {

    BRAND_COLOR: _ColorPalette2.default.brandColor,

    MALE_COLOR: _ColorPalette2.default.lighterBlueColor,
    FEMALE_COLOR: _ColorPalette2.default.pinkColor,
    UNKNOWN_GENDER_COLOR: _ColorPalette2.default.neutralColor,

    POSITIVE_COLOR: _ColorPalette2.default.positiveColor,
    NEGATIVE_COLOR: _ColorPalette2.default.negativeColor,
    NEUTRAL_COLOR: _ColorPalette2.default.neutralColor,

    GRID_COLOR: _ColorPalette2.default.placeholderColor,
    BACKGROUND_COLOR: _ColorPalette2.default.darkBlueColor,
    LABEL_COLOR: _ColorPalette2.default.placeholderColor,
    VALUE_COLOR: _ColorPalette2.default.whiteColor,

    OVERLAY_COLOR: _ColorPalette2.default.whiteColor,
    KNOCKOUT_COLOR: _ColorPalette2.default.fillBlueColor,

    palettes: {
      BRIGHT: [_ColorPalette2.default.bright0Color, _ColorPalette2.default.bright1Color, _ColorPalette2.default.bright2Color, _ColorPalette2.default.bright3Color, _ColorPalette2.default.bright4Color, _ColorPalette2.default.bright5Color, _ColorPalette2.default.bright6Color, _ColorPalette2.default.bright7Color],
      MONOCHROMATIC: [_ColorPalette2.default.mono0Color, _ColorPalette2.default.mono1Color, _ColorPalette2.default.mono2Color, _ColorPalette2.default.mono3Color, _ColorPalette2.default.mono4Color]
    }

  },

  report: {

    BRAND_COLOR: _ColorPalette2.default.brandColor,

    MALE_COLOR: _ColorPalette2.default.lighterBlueColor,
    FEMALE_COLOR: _ColorPalette2.default.pinkColor,
    UNKNOWN_GENDER_COLOR: _ColorPalette2.default.neutralColor,

    POSITIVE_COLOR: _ColorPalette2.default.positiveDarkColor,
    NEGATIVE_COLOR: _ColorPalette2.default.negativeDarkColor,
    NEUTRAL_COLOR: _ColorPalette2.default.neutralColor,

    GRID_COLOR: _ColorPalette2.default.grayLightColor,
    BACKGROUND_COLOR: _ColorPalette2.default.grayLighterColor,
    LABEL_COLOR: _ColorPalette2.default.grayColor,
    VALUE_COLOR: _ColorPalette2.default.grayColor,

    OVERLAY_COLOR: _ColorPalette2.default.grayColor,
    KNOCKOUT_COLOR: _ColorPalette2.default.whiteColor,

    palettes: {
      BRIGHT: [_ColorPalette2.default.bright0Color, _ColorPalette2.default.bright1Color, _ColorPalette2.default.bright2Color, _ColorPalette2.default.bright3Color, _ColorPalette2.default.bright4Color, _ColorPalette2.default.bright5Color, _ColorPalette2.default.bright6Color, _ColorPalette2.default.bright7Color],
      MONOCHROMATIC: [_ColorPalette2.default.mono0Color, _ColorPalette2.default.mono1Color, _ColorPalette2.default.mono2Color, _ColorPalette2.default.mono3Color, _ColorPalette2.default.mono4Color]
    }

  }

};
